src/js/components/chat-input.jsx

```diff
@@ -1,61 +1,71 @@
 import React, { Component, PropTypes } from 'react';
 import { connect } from 'react-redux';
 import isMobile from 'ismobilejs';
+import bindAll from 'lodash.bindall';
 
-import { sendMessage, resetUnreadCount } from '../services/conversation-service';
-import { store } from '../stores/app-store';
+import { sendMessage, resetUnreadCount } from '../services/conversation';
 
 import { ImageUpload } from './image-upload';
 
-function checkAndResetUnreadCount() {
-    if (store.getState().conversation.unreadCount > 0) {
-        resetUnreadCount();
-    }
-}
+
 
 export class ChatInputComponent extends Component {
 
     static propTypes = {
         accentColor: PropTypes.string,
         imageUploadEnabled: PropTypes.bool.isRequired,
         inputPlaceholderText: PropTypes.string.isRequired,
-        sendButtonText: PropTypes.string.isRequired
+        sendButtonText: PropTypes.string.isRequired,
+        unreadCount: PropTypes.number.isRequired,
+        dispatch: PropTypes.func.isRequired
+    };
+
+    state = {
+        text: ''
     };
 
     constructor(...args) {
         super(...args);
-
-        this.state = {
-            text: ''
-        };
-
-        this.onChange = this.onChange.bind(this);
-        this.onSendMessage = this.onSendMessage.bind(this);
+        bindAll(this, [
```

Comment:
I think bindAll accepts rest parameters, so you could send separate string args instead of an array. But maybe you prefer on their own lines? If so this is fine.